tool/clean/clean_security_group/clean_security_group.go

```diff
@@ -0,0 +1,444 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/aws/amazon-cloudwatch-agent/tool/clean"
+)
+
+type ec2Client interface {
+	DescribeSecurityGroups(ctx context.Context, params *ec2.DescribeSecurityGroupsInput, optFns ...func(*ec2.Options)) (*ec2.DescribeSecurityGroupsOutput, error)
+	DescribeNetworkInterfaces(ctx context.Context, params *ec2.DescribeNetworkInterfacesInput, optFns ...func(*ec2.Options)) (*ec2.DescribeNetworkInterfacesOutput, error)
+	DeleteSecurityGroup(ctx context.Context, params *ec2.DeleteSecurityGroupInput, optFns ...func(*ec2.Options)) (*ec2.DeleteSecurityGroupOutput, error)
+	RevokeSecurityGroupIngress(ctx context.Context, params *ec2.RevokeSecurityGroupIngressInput, optFns ...func(*ec2.Options)) (*ec2.RevokeSecurityGroupIngressOutput, error)
+	RevokeSecurityGroupEgress(ctx context.Context, params *ec2.RevokeSecurityGroupEgressInput, optFns ...func(*ec2.Options)) (*ec2.RevokeSecurityGroupEgressOutput, error)
+}
+
+const (
+	SecurityGroupProcessChanSize = 500
+)
+
+// Config holds the application configuration
+type Config struct {
+	ageThreshold  time.Duration
+	numWorkers    int
+	exceptionList []string
+	dryRun        bool
+	skipVpcSGs    bool
+	skipWithRules bool
+}
+
+// Global configuration
+var (
+	cfg Config
+)
+
+func init() {
+	// Set default configuration
+	cfg = Config{
+		ageThreshold:  1 * clean.KeepDurationOneDay,
+		numWorkers:    30,
+		exceptionList: []string{"default"},
+		dryRun:        true,
+		skipVpcSGs:    false,
+		skipWithRules: false,
+	}
+}
+
+func main() {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
+	defer cancel()
+
+	// Parse command line flags
+	flag.BoolVar(&cfg.dryRun, "dry-run", true, "Enable dry-run mode (no actual deletion)")
+	flag.DurationVar(&cfg.ageThreshold, "age", 1*clean.KeepDurationOneDay, "Age threshold for security groups (e.g. 24h)")
+	flag.BoolVar(&cfg.skipVpcSGs, "skip-vpc", false, "Skip security groups associated with VPCs")
+	flag.BoolVar(&cfg.skipWithRules, "skip-with-rules", false, "Skip security groups that have ingress or egress rules")
+	flag.Parse()
+
+	// Load AWS configuration
+	awsCfg, err := loadAWSConfig(ctx)
+	if err != nil {
+		log.Fatalf("Error loading AWS config: %v", err)
+	}
+
+	// Create EC2 client
+	client := ec2.NewFromConfig(awsCfg)
+
+	log.Printf("🔍 Searching for unused Security Groups older than %v in %s region\n",
+		cfg.ageThreshold, awsCfg.Region)
+
+	// Delete old security groups
+	deletedGroups := deleteUnusedSecurityGroups(ctx, client)
```

Comment:
Not a blocker but maybe add some error handling to the funciton and make it return an erro so we know why the security group wasn't able delete the unused security group.
```suggestion
	    deletedGroups, err := deleteUnusedSecurityGroups(ctx, client)
    if err != nil {
        log.Fatalf("Error deleting unused security groups: %v", err)
    }
    log.Printf("Total security groups deleted: %d", len(deletedGroups))
```